lib/docsplit_processor/paperclip/pdf_extractor.rb

```diff
@@ -0,0 +1,43 @@
+require 'paperclip'
+require 'docsplit'
+
+module Paperclip
+  class PdfExtractor < Processor
+    def make
+      if original_cv_is_pdf?
+        File.open(input_file_path)
+      else
+        Docsplit.extract_pdf(input_file_path, output: destination_dir)
+        File.open(output_file_path)
+      end
+    rescue StandardError
+      raise Paperclip::Error, "Error converting '#{input_file_path}' to pdf"
+    end
+
+    private
+
+    def original_cv_is_pdf?
+      input_file_path.match(/pdf$/)
+    end
```

Comment:
could use something more sophisticated like `https://github.com/blackwinter/ruby-filemagic` of just good old `file` to check the mime type but this looks fine

Reply:
unless someone tries to upload something doddgy with an extension of pdf of course 🤔 but then the conversion would fail anyway. I presume we throw away anything that doest convert anyway so it wouldnt pose any sort of security risk if someone was trying to upload something they shouldnt be

Reply:
The other processor gems that I looked at all use filemagic. The only reason I didn't was pure laziness. Maybe at some point someone can tighten this up a bit.